packages/server/lib/browsers/utils.ts

```diff
@@ -182,44 +182,93 @@ function extendLaunchOptionsFromPlugins (launchOptions, pluginConfigResult, opti
   return launchOptions
 }
 
-const getBrowsers = () => {
-  debug('getBrowsers')
+const wkBrowserVersionRe = /BROWSER_VERSION = \'(?<version>[^']+)\'/gm
 
-  return launcher.detect()
-  .then((browsers: FoundBrowser[] = []) => {
-    let majorVersion
+const getWebKitBrowserVersion = async () => {
+  try {
+    // this seems to be the only way to accurately capture the WebKit version - it's not exported, and invoking the webkit binary with `--version` does not give the correct result
+    // after launching the browser, this is available at browser.version(), but we don't have a browser instance til later
+    const pwCorePath = path.dirname(require.resolve('playwright-core', { paths: [process.cwd()] }))
+    const wkBrowserPath = path.join(pwCorePath, 'lib', 'server', 'webkit', 'wkBrowser.js')
+    const wkBrowserContents = await fs.readFile(wkBrowserPath)
+    const result = wkBrowserVersionRe.exec(wkBrowserContents)
 
-    debug('found browsers %o', { browsers })
+    if (!result || !result.groups!.version) return '0'
 
-    if (!process.versions.electron) {
-      debug('not in electron, skipping adding electron browser')
+    return result.groups!.version
+  } catch (err) {
+    debug('Error detecting WebKit browser version %o', err)
 
-      return browsers
-    }
-
-    // @ts-ignore
-    const version = process.versions.chrome || ''
+    return '0'
+  }
+}
 
-    if (version) {
-      majorVersion = getMajorVersion(version)
-    }
+const getWebKitBrowser = async () => {
+  try {
+    const modulePath = require.resolve('playwright-webkit', { paths: [process.cwd()] })
+    const mod = require(modulePath) as typeof import('playwright-webkit')
+    const version = await getWebKitBrowserVersion()
 
-    const electronBrowser: FoundBrowser = {
-      name: 'electron',
+    const browser: FoundBrowser = {
+      name: 'webkit',
       channel: 'stable',
-      family: 'chromium',
-      displayName: 'Electron',
+      family: 'webkit',
+      displayName: 'WebKit',
       version,
-      path: '',
-      majorVersion,
-      info: 'Electron is the default browser that comes with Cypress. This is the default browser that runs in headless mode. Selecting this browser is useful when debugging. The version number indicates the underlying Chromium version that Electron uses.',
+      path: mod.webkit.executablePath(),
+      majorVersion: version.split('.')[0],
```

Comment:
Nit, but should we just use the `major` function from the `semver` package we've already got for these in general? Eg

```js
import semver from 'semver'

const majorVersion = semver.major(version)
```

We could even change `getWebKitBrowserVersion` to return the result of this:


```js
return semver.major(result?.groups?.version) ?? '0.0.0'
```

Reply:
I'm a fan of not using dependencies wherever reasonable. AFAIK we only use `semver` in one place in the server as it is.

Reply:
Agree on this general sentiment - if we only use `semver` in one place, maybe we should just drop it (and change that one place to not use it).